### PR TITLE
Make recruitment of heroes more random

### LINKED ISSUE #1315: Possibility to recruit all the heroes of the same class as the original castle in succession
In fheroes2, it is possible to recruit all the heroes of the same class as the original castle in succession.
This never happens in the original game.

In this example, all the Warlock class heroes have been recruited consecutively.
And the ninth Warlock class hero is even available at the tavern.

Recruit all the heroes of the same class 01

Recruit all the heroes of the same class 02

### PR DESCRIPTION
**Pull request by oleg-derevenetz**
fix #1315

### CODE REVIEW

File: src/fheroes2/kingdom/kingdom.cpp [hunk 1]
```diff
@@ -451,7 +451,7 @@ Recruits & Kingdom::GetRecruits( void )
 {
     // update hero1
     if ( Heroes::UNKNOWN == recruits.GetID1() || ( recruits.GetHero1() && !recruits.GetHero1()->isFreeman() ) )
-        recruits.SetHero1( world.GetFreemanHeroes( GetRace() ) );
+        recruits.SetHero1( world.GetFreemanHeroes() );
```
Comment (idshibanov) [inline src/fheroes2/kingdom/kingdom.cpp:454]: I think actual condition here should be a bit more complicated. When new game is started 1st recruit is guaranteed to be of the same race. Once bought it is indeed random.

**Comment by oleg-derevenetz** [comment]
Hi @idshibanov

> When new game is started 1st recruit is guaranteed to be of the same race.

Hmm, I thought that such guarantees exist only for starting (already hired) hero... Well, then yes, logic should be more complicated. I'll rework this.

**Review by idshibanov — APPROVED** [review]

File: src/fheroes2/kingdom/kingdom.cpp [hunk 2]
```diff
@@ -449,9 +449,11 @@ u32 Kingdom::GetLostTownDays( void ) const
 
 Recruits & Kingdom::GetRecruits( void )
 {
+    const bool preferNative = recruits.GetID1() == Heroes::UNKNOWN && recruits.GetID2() == Heroes::UNKNOWN;
```
Comment (ihhub) [inline src/fheroes2/kingdom/kingdom.cpp:452]: Please put this variable under the if-condition below as it effects only that code.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @oleg-derevenetz , only one small optimization is required. Could you please take a look?

**Comment by ihhub** [comment]
@oleg-derevenetz , good job! A bunch of thanks for fixing this bug!

**Pull request merged**